# Import libraries and the dataset.

In [8]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [9]:
df = pd.read_csv('train.csv')

In [10]:
df.shape

(9800, 18)

In [11]:
df.head()

Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2017-152156  08/11/2017  11/11/2017    Second Class    CG-12520   
1       2  CA-2017-152156  08/11/2017  11/11/2017    Second Class    CG-12520   
2       3  CA-2017-138688  12/06/2017  16/06/2017    Second Class    DV-13045   
3       4  US-2016-108966  11/10/2016  18/10/2016  Standard Class    SO-20335   
4       5  US-2016-108966  11/10/2016  18/10/2016  Standard Class    SO-20335   

     Customer Name    Segment        Country             City       State  \
0      Claire Gute   Consumer  United States        Henderson    Kentucky   
1      Claire Gute   Consumer  United States        Henderson    Kentucky   
2  Darrin Van Huff  Corporate  United States      Los Angeles  California   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale     Florida   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale     Florida   

   Postal Code Region       Product ID         Category Sub-Category  \
0      42420.0  South  FUR-BO-10001798        Furniture    Bookcases   
1      42420.0  South  FUR-CH-10000454        Furniture       Chairs   
2      90036.0   West  OFF-LA-10000240  Office Supplies       Labels   
3      33311.0  South  FUR-TA-10000577        Furniture       Tables   
4      33311.0  South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  
0                  Bush Somerset Collection Bookcase  261.9600  
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400  
2  Self-Adhesive Address Labels for Typewriters b...   14.6200  
3      Bretford CR4500 Series Slim Rectangular Table  957.5775  
4                     Eldon Fold 'N Roll Cart System   22.3680

In [12]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales'],
      dtype='object')

In [13]:
df.drop(['Row ID','Order ID'],axis=1,inplace=True)

# Creating day month and year columns.

In [14]:
df['Order Day'] = df['Order Date'].str.split('/').str[0]
df['Order Month'] = df['Order Date'].str.split('/').str[1]
df['Order Year'] = df['Order Date'].str.split('/').str[2]

In [15]:
df['Order Day'] = df['Order Day'].astype(int)
df['Order Month'] = df['Order Month'].astype(int)
df['Order Year'] = df['Order Year'].astype(int)

In [16]:
df.head()

Order Date   Ship Date       Ship Mode Customer ID    Customer Name  \
0  08/11/2017  11/11/2017    Second Class    CG-12520      Claire Gute   
1  08/11/2017  11/11/2017    Second Class    CG-12520      Claire Gute   
2  12/06/2017  16/06/2017    Second Class    DV-13045  Darrin Van Huff   
3  11/10/2016  18/10/2016  Standard Class    SO-20335   Sean O'Donnell   
4  11/10/2016  18/10/2016  Standard Class    SO-20335   Sean O'Donnell   

     Segment        Country             City       State  Postal Code Region  \
0   Consumer  United States        Henderson    Kentucky      42420.0  South   
1   Consumer  United States        Henderson    Kentucky      42420.0  South   
2  Corporate  United States      Los Angeles  California      90036.0   West   
3   Consumer  United States  Fort Lauderdale     Florida      33311.0  South   
4   Consumer  United States  Fort Lauderdale     Florida      33311.0  South   

        Product ID         Category Sub-Category  \
0  FUR-BO-10001798        Furniture    Bookcases   
1  FUR-CH-10000454        Furniture       Chairs   
2  OFF-LA-10000240  Office Supplies       Labels   
3  FUR-TA-10000577        Furniture       Tables   
4  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Order Day  \
0                  Bush Somerset Collection Bookcase  261.9600          8   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400          8   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         12   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         11   
4                     Eldon Fold 'N Roll Cart System   22.3680         11   

   Order Month  Order Year  
0           11        2017  
1           11        2017  
2            6        2017  
3           10        2016  
4           10        2016

# Creating shipping date, month and year.

In [19]:
df['Ship Day'] = df['Ship Date'].str.split('/').str[0]
df['Ship Month'] = df['Ship Date'].str.split('/').str[1]
df['Ship Year'] = df['Ship Date'].str.split('/').str[2]

In [20]:
df['Ship Day'] = df['Ship Day'].astype(int)
df['Ship Month'] = df['Ship Month'].astype(int)
df['Ship Year'] = df['Ship Year'].astype(int)

In [21]:
df.head()

Order Date   Ship Date       Ship Mode Customer ID    Customer Name  \
0  08/11/2017  11/11/2017    Second Class    CG-12520      Claire Gute   
1  08/11/2017  11/11/2017    Second Class    CG-12520      Claire Gute   
2  12/06/2017  16/06/2017    Second Class    DV-13045  Darrin Van Huff   
3  11/10/2016  18/10/2016  Standard Class    SO-20335   Sean O'Donnell   
4  11/10/2016  18/10/2016  Standard Class    SO-20335   Sean O'Donnell   

     Segment        Country             City       State  Postal Code  \
0   Consumer  United States        Henderson    Kentucky      42420.0   
1   Consumer  United States        Henderson    Kentucky      42420.0   
2  Corporate  United States      Los Angeles  California      90036.0   
3   Consumer  United States  Fort Lauderdale     Florida      33311.0   
4   Consumer  United States  Fort Lauderdale     Florida      33311.0   

     ...             Category Sub-Category  \
0    ...            Furniture    Bookcases   
1    ...            Furniture       Chairs   
2    ...      Office Supplies       Labels   
3    ...            Furniture       Tables   
4    ...      Office Supplies      Storage   

                                        Product Name     Sales Order Day  \
0                  Bush Somerset Collection Bookcase  261.9600         8   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         8   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200        12   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775        11   
4                     Eldon Fold 'N Roll Cart System   22.3680        11   

   Order Month  Order Year  Ship Day  Ship Month  Ship Year  
0           11        2017        11          11       2017  
1           11        2017        11          11       2017  
2            6        2017        16           6       2017  
3           10        2016        18          10       2016  
4           10        2016        18          10       2016  

[5 rows x 22 columns]

In [22]:
df.drop(['Order Date','Ship Date'],axis=1 ,inplace=True)

In [23]:
df.head()

Ship Mode Customer ID    Customer Name    Segment        Country  \
0    Second Class    CG-12520      Claire Gute   Consumer  United States   
1    Second Class    CG-12520      Claire Gute   Consumer  United States   
2    Second Class    DV-13045  Darrin Van Huff  Corporate  United States   
3  Standard Class    SO-20335   Sean O'Donnell   Consumer  United States   
4  Standard Class    SO-20335   Sean O'Donnell   Consumer  United States   

              City       State  Postal Code Region       Product ID  \
0        Henderson    Kentucky      42420.0  South  FUR-BO-10001798   
1        Henderson    Kentucky      42420.0  South  FUR-CH-10000454   
2      Los Angeles  California      90036.0   West  OFF-LA-10000240   
3  Fort Lauderdale     Florida      33311.0  South  FUR-TA-10000577   
4  Fort Lauderdale     Florida      33311.0  South  OFF-ST-10000760   

          Category Sub-Category  \
0        Furniture    Bookcases   
1        Furniture       Chairs   
2  Office Supplies       Labels   
3        Furniture       Tables   
4  Office Supplies      Storage   

                                        Product Name     Sales  Order Day  \
0                  Bush Somerset Collection Bookcase  261.9600          8   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400          8   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         12   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         11   
4                     Eldon Fold 'N Roll Cart System   22.3680         11   

   Order Month  Order Year  Ship Day  Ship Month  Ship Year  
0           11        2017        11          11       2017  
1           11        2017        11          11       2017  
2            6        2017        16           6       2017  
3           10        2016        18          10       2016  
4           10        2016        18          10       2016

# What was the best month for sales in 2018? How much was earned that month?

In [34]:
new_df=df[df['Order Year']==2018]

In [40]:
new_df.groupby('Order Month').sum().sort_values('Sales',ascending=False)

Postal Code        Sales  Order Day  Order Year  Ship Day  \
Order Month                                                              
11            25171742.0  117938.1550       6746      920208      7009   
9             24711545.0   86152.8880       6625      914154      7278   
12            25567545.0   83030.3888       6120      926262      6667   
10            16394453.0   77448.1312       4539      593292      4536   
8             13236311.0   62837.8480       3672      431852      3669   
3             13925675.0   58863.4128       4052      478266      4125   
6             13140015.0   48190.7277       3719      474230      3718   
7             11788820.0   44825.1040       3583      450014      3419   
5             14034388.0   43825.9822       3758      484320      3722   
1              8721759.0   43476.4740       2544      288574      2412   
4             10005651.0   35541.9101       3266      403600      3108   
2              6515796.0   19920.9974       1464      209872      1665   

             Ship Month  Ship Year  
Order Month                         
11                 5065     920208  
9                  4111     914154  
12                 5046     926304  
10                 2978     593292  
8                  1740     431852  
3                   736     478266  
6                  1442     474230  
7                  1596     450014  
5                  1230     484320  
1                   166     288574  
4                   831     403600  
2                   217     209872

Answer : November month had the best sales in year 2018.
         the earned amount was  $117,938.1550 in November

# What State had the highest number of sales?

In [63]:
sales_df = new_df.groupby('State').sum()
sales_df

Postal Code        Sales  Order Day  Order Month  \
State                                                                    
Alabama                  570606.0    1828.2500        263          113   
Arizona                 5881897.0   11124.1930       1098          531   
Arkansas                1373856.0    2707.6400        318          181   
California             60221339.0  144666.7685       9768         4985   
Colorado                4652704.0   10033.2930        755          518   
Connecticut              222615.0    5307.2100        489          219   
Delaware                 375073.0   13754.9830        260          129   
District of Columbia      40032.0      77.7600         24            2   
Florida                 4140781.0   26444.7155       1813          971   
Georgia                 1864926.0   18873.9300       1072          391   
Idaho                    667875.0    1233.5480        140           66   
Illinois               10207090.0   23848.0430       2768         1202   
Indiana                 2253872.0   13685.4300        786          389   
Iowa                     613018.0     580.2100        256           64   
Kansas                   399906.0     733.1500         82           48   
Kentucky                2044476.0   15516.8400        874          476   
Louisiana                987876.0    5501.5500        267          115   
Maryland                 780492.0    9451.6000        582          277   
Massachusetts             80216.0    8143.0760        532          348   
Michigan                3248248.0   25833.6490        881          509   
Minnesota               2113181.0    6728.2500        554          318   
Mississippi              508261.0    2997.2400        153          106   
Missouri                1693706.0    9350.8000        301          186   
Montana                  297221.0    4229.2800        113           13   
Nebraska                 681342.0    3579.3500        140           91   
Nevada                   534506.0    3138.1580        115           37   
New Hampshire             22866.0    1509.4300         75           54   
New Jersey               306932.0    8675.6800        524          310   
New Mexico              1927595.0    2818.5580        350          181   
New York                3613521.0   93772.1670       5613         2900   
North Carolina          2330815.0   23019.6290       1158          648   
North Dakota             406721.0     919.9100        156           49   
Ohio                    6847080.0   21372.7600       2397         1195   
Oklahoma                1254007.0    6225.9100        388          125   
Oregon                  2921152.0    2886.5930        494          256   
Pennsylvania            3756597.0   42688.3060       2979         1559   
Rhode Island              58164.0    3324.1800        325          181   
South Carolina           235281.0    1560.7500        126           73   
South Dakota             457122.0    1153.4100        156           60   
Tennessee               3057696.0   16113.8230       1196          660   
Texas                  24329849.0   43421.7582       4827         2350   
Utah                    1264857.0    2460.6380        226          133   
Vermont                       0.0     842.2100         81           16   
Virginia                1002150.0    7600.2000        727          296   
Washington             21101318.0   65539.8960       3204         1630   
West Virginia            104012.0    1209.8240         75           28   
Wisconsin               1760850.0    5567.4700        607          297   

                      Order Year  Ship Day  Ship Month  Ship Year  
State                                                              
Alabama                    32288       331         113      32288  
Arizona                   139242      1172         537     139242  
Arkansas                   38342       395         181      38342  
California               1315736     10183        4980 

In [64]:
sales_df = sales_df.sort_values('Sales', ascending = False)
sales_df.head(5)

Postal Code        Sales  Order Day  Order Month  Order Year  \
State                                                                        
California     60221339.0  144666.7685       9768         4985     1315736   
New York        3613521.0   93772.1670       5613         2900      704282   
Washington     21101318.0   65539.8960       3204         1630      433870   
Texas          24329849.0   43421.7582       4827         2350      639706   
Pennsylvania    3756597.0   42688.3060       2979         1559      397546   

              Ship Day  Ship Month  Ship Year  
State                                          
California       10183        4980    1315742  
New York          5921        2854     704289  
Washington        3101        1624     433873  
Texas             4946        2291     639714  
Pennsylvania      3062        1582     397546

Answer : California has the most sales of amount $144666.7685

# What product was sold the most?

In [68]:
product_df = new_df.groupby('Product Name').sum()

In [70]:
product_df.sort_values('Sales',ascending=False).head()

Postal Code      Sales  \
Product Name                                                                 
Canon imageCLASS 2200 Advanced Copier                  127850.0  35699.898   
Martin Yale Chadless Opener Electric Letter Opener     235666.0  11825.902   
GBC DocuBind TL300 Electric Binding System             152770.0  10943.278   
Hewlett Packard LaserJet 3310 Copier                   273623.0   9239.846   
Samsung Galaxy Mega 6.3                                159546.0   9239.780   

                                                    Order Day  Order Month  \
Product Name                                                                 
Canon imageCLASS 2200 Advanced Copier                      62           24   
Martin Yale Chadless Opener Electric Letter Opener         67           25   
GBC DocuBind TL300 Electric Binding System                 61           39   
Hewlett Packard LaserJet 3310 Copier                       92           25   
Samsung Galaxy Mega 6.3                                    81           31   

                                                    Order Year  Ship Day  \
Product Name                                                               
Canon imageCLASS 2200 Advanced Copier                     6054        71   
Martin Yale Chadless Opener Electric Letter Opener        8072        85   
GBC DocuBind TL300 Electric Binding System               10090        85   
Hewlett Packard LaserJet 3310 Copier                      8072        64   
Samsung Galaxy Mega 6.3                                   8072        91   

                                                    Ship Month  Ship Year  
Product Name                                                               
Canon imageCLASS 2200 Advanced Copier                       24       6054  
Martin Yale Chadless Opener Electric Letter Opener          25       8072  
GBC DocuBind TL300 Electric Binding System                  39      10090  
Hewlett Packard LaserJet 3310 Copier                        26       8072  
Samsung Galaxy Mega 6.3                                     31       8072

Answer : Canon imageCLASS 2200 Advanced Copier was sold the most in 2018.

In [71]:
df['Ship Mode'].unique()

array(['Second Class', 'Standard Class', 'First Class', 'Same Day'],
      dtype=object)

# Performing Encoding.

In [72]:
df['Ship Mode'] = df['Ship Mode'].map({'Second Class':3, 'Standard Class':2, 'First Class':1, 'Same Day':0})

In [73]:
df['Ship Mode'] = df['Ship Mode'].astype(int)

In [74]:
df.head()

Ship Mode Customer ID    Customer Name    Segment        Country  \
0          3    CG-12520      Claire Gute   Consumer  United States   
1          3    CG-12520      Claire Gute   Consumer  United States   
2          3    DV-13045  Darrin Van Huff  Corporate  United States   
3          2    SO-20335   Sean O'Donnell   Consumer  United States   
4          2    SO-20335   Sean O'Donnell   Consumer  United States   

              City       State  Postal Code Region       Product ID  \
0        Henderson    Kentucky      42420.0  South  FUR-BO-10001798   
1        Henderson    Kentucky      42420.0  South  FUR-CH-10000454   
2      Los Angeles  California      90036.0   West  OFF-LA-10000240   
3  Fort Lauderdale     Florida      33311.0  South  FUR-TA-10000577   
4  Fort Lauderdale     Florida      33311.0  South  OFF-ST-10000760   

          Category Sub-Category  \
0        Furniture    Bookcases   
1        Furniture       Chairs   
2  Office Supplies       Labels   
3        Furniture       Tables   
4  Office Supplies      Storage   

                                        Product Name     Sales  Order Day  \
0                  Bush Somerset Collection Bookcase  261.9600          8   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400          8   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         12   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         11   
4                     Eldon Fold 'N Roll Cart System   22.3680         11   

   Order Month  Order Year  Ship Day  Ship Month  Ship Year  
0           11        2017        11          11       2017  
1           11        2017        11          11       2017  
2            6        2017        16           6       2017  
3           10        2016        18          10       2016  
4           10        2016        18          10       2016

In [75]:
#Checking null values.
df.isnull().sum()

Ship Mode         0
Customer ID       0
Customer Name     0
Segment           0
Country           0
City              0
State             0
Postal Code      11
Region            0
Product ID        0
Category          0
Sub-Category      0
Product Name      0
Sales             0
Order Day         0
Order Month       0
Order Year        0
Ship Day          0
Ship Month        0
Ship Year         0
dtype: int64

In [76]:
df['Postal Code'].describe()

count     9789.000000
mean     55273.322403
std      32041.223413
min       1040.000000
25%      23223.000000
50%      58103.000000
75%      90008.000000
max      99301.000000
Name: Postal Code, dtype: float64

In [79]:
#filling null values with mean.
df['Postal Code'] = df['Postal Code'].fillna(df['Postal Code'].mean()) 

In [80]:
df.isnull().sum()

Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Order Day        0
Order Month      0
Order Year       0
Ship Day         0
Ship Month       0
Ship Year        0
dtype: int64

In [81]:
#Dropping columns.
df.drop(['Customer ID','Customer Name'],axis=1, inplace = True)

In [82]:
df.head()

Ship Mode    Segment        Country             City       State  \
0          3   Consumer  United States        Henderson    Kentucky   
1          3   Consumer  United States        Henderson    Kentucky   
2          3  Corporate  United States      Los Angeles  California   
3          2   Consumer  United States  Fort Lauderdale     Florida   
4          2   Consumer  United States  Fort Lauderdale     Florida   

   Postal Code Region       Product ID         Category Sub-Category  \
0      42420.0  South  FUR-BO-10001798        Furniture    Bookcases   
1      42420.0  South  FUR-CH-10000454        Furniture       Chairs   
2      90036.0   West  OFF-LA-10000240  Office Supplies       Labels   
3      33311.0  South  FUR-TA-10000577        Furniture       Tables   
4      33311.0  South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Order Day  \
0                  Bush Somerset Collection Bookcase  261.9600          8   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400          8   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         12   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         11   
4                     Eldon Fold 'N Roll Cart System   22.3680         11   

   Order Month  Order Year  Ship Day  Ship Month  Ship Year  
0           11        2017        11          11       2017  
1           11        2017        11          11       2017  
2            6        2017        16           6       2017  
3           10        2016        18          10       2016  
4           10        2016        18          10       2016

In [83]:
df['Segment'].unique()

array(['Consumer', 'Corporate', 'Home Office'], dtype=object)

In [84]:
df['Segment'] = df['Segment'].map({'Consumer':0, 'Corporate':1, 'Home Office':2})

In [85]:
df['Segment'] = df['Segment'].astype(int)

In [86]:
df.head()

Ship Mode  Segment        Country             City       State  \
0          3        0  United States        Henderson    Kentucky   
1          3        0  United States        Henderson    Kentucky   
2          3        1  United States      Los Angeles  California   
3          2        0  United States  Fort Lauderdale     Florida   
4          2        0  United States  Fort Lauderdale     Florida   

   Postal Code Region       Product ID         Category Sub-Category  \
0      42420.0  South  FUR-BO-10001798        Furniture    Bookcases   
1      42420.0  South  FUR-CH-10000454        Furniture       Chairs   
2      90036.0   West  OFF-LA-10000240  Office Supplies       Labels   
3      33311.0  South  FUR-TA-10000577        Furniture       Tables   
4      33311.0  South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Order Day  \
0                  Bush Somerset Collection Bookcase  261.9600          8   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400          8   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         12   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         11   
4                     Eldon Fold 'N Roll Cart System   22.3680         11   

   Order Month  Order Year  Ship Day  Ship Month  Ship Year  
0           11        2017        11          11       2017  
1           11        2017        11          11       2017  
2            6        2017        16           6       2017  
3           10        2016        18          10       2016  
4           10        2016        18          10       2016

In [87]:
df['Country'].unique()

array(['United States'], dtype=object)

In [88]:
#assigning 1 as their is only one country.
df['Country'] = 1

In [89]:
df['Country'] = df['Country'].astype(int)

In [90]:
df.head()

Ship Mode  Segment  Country             City       State  Postal Code  \
0          3        0        1        Henderson    Kentucky      42420.0   
1          3        0        1        Henderson    Kentucky      42420.0   
2          3        1        1      Los Angeles  California      90036.0   
3          2        0        1  Fort Lauderdale     Florida      33311.0   
4          2        0        1  Fort Lauderdale     Florida      33311.0   

  Region       Product ID         Category Sub-Category  \
0  South  FUR-BO-10001798        Furniture    Bookcases   
1  South  FUR-CH-10000454        Furniture       Chairs   
2   West  OFF-LA-10000240  Office Supplies       Labels   
3  South  FUR-TA-10000577        Furniture       Tables   
4  South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Order Day  \
0                  Bush Somerset Collection Bookcase  261.9600          8   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400          8   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         12   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         11   
4                     Eldon Fold 'N Roll Cart System   22.3680         11   

   Order Month  Order Year  Ship Day  Ship Month  Ship Year  
0           11        2017        11          11       2017  
1           11        2017        11          11       2017  
2            6        2017        16           6       2017  
3           10        2016        18          10       2016  
4           10        2016        18          10       2016

In [91]:
df.drop(['City','State','Product ID','Sub-Category','Product Name'], axis=1, inplace=True)

In [92]:
df.head()

Ship Mode  Segment  Country  Postal Code Region         Category     Sales  \
0          3        0        1      42420.0  South        Furniture  261.9600   
1          3        0        1      42420.0  South        Furniture  731.9400   
2          3        1        1      90036.0   West  Office Supplies   14.6200   
3          2        0        1      33311.0  South        Furniture  957.5775   
4          2        0        1      33311.0  South  Office Supplies   22.3680   

   Order Day  Order Month  Order Year  Ship Day  Ship Month  Ship Year  
0          8           11        2017        11          11       2017  
1          8           11        2017        11          11       2017  
2         12            6        2017        16           6       2017  
3         11           10        2016        18          10       2016  
4         11           10        2016        18          10       2016

In [93]:
df['Region'].unique()

array(['South', 'West', 'Central', 'East'], dtype=object)

In [94]:
df['Region'] = df['Region'].map({'South':1,'West':2,'Central':3,'East':4})

In [95]:
df['Region'] = df['Region'].astype(int)

In [96]:
df['Category'].unique()

array(['Furniture', 'Office Supplies', 'Technology'], dtype=object)

In [97]:
df['Category'] = df['Category'].map({'Furniture':1,'Office Supplies':2,'Technology':3})

In [98]:
df['Category'] = df['Category'].astype(int)

In [99]:
df.head()

Ship Mode  Segment  Country  Postal Code  Region  Category     Sales  \
0          3        0        1      42420.0       1         1  261.9600   
1          3        0        1      42420.0       1         1  731.9400   
2          3        1        1      90036.0       2         2   14.6200   
3          2        0        1      33311.0       1         1  957.5775   
4          2        0        1      33311.0       1         2   22.3680   

   Order Day  Order Month  Order Year  Ship Day  Ship Month  Ship Year  
0          8           11        2017        11          11       2017  
1          8           11        2017        11          11       2017  
2         12            6        2017        16           6       2017  
3         11           10        2016        18          10       2016  
4         11           10        2016        18          10       2016

In [100]:
df.dtypes

Ship Mode        int32
Segment          int32
Country          int32
Postal Code    float64
Region           int32
Category         int32
Sales          float64
Order Day        int32
Order Month      int32
Order Year       int32
Ship Day         int32
Ship Month       int32
Ship Year        int32
dtype: object

In [101]:
df.corr()

Ship Mode   Segment  Country  Postal Code    Region  Category  \
Ship Mode     1.000000 -0.008404      NaN    -0.004952 -0.011749 -0.002217   
Segment      -0.008404  1.000000      NaN    -0.009517  0.004708  0.009412   
Country            NaN       NaN      NaN          NaN       NaN       NaN   
Postal Code  -0.004952 -0.009517      NaN     1.000000 -0.457813 -0.000795   
Region       -0.011749  0.004708      NaN    -0.457813  1.000000  0.002441   
Category     -0.002217  0.009412      NaN    -0.000795  0.002441  1.000000   
Sales         0.002101  0.010924      NaN    -0.024002  0.000070  0.041375   
Order Day     0.020068  0.003919      NaN     0.017929 -0.011606 -0.014295   
Order Month  -0.010100 -0.022306      NaN     0.021646  0.017303 -0.012946   
Order Year   -0.026663  0.040080      NaN     0.006415  0.012116  0.004313   
Ship Day      0.031236  0.002013      NaN     0.029721 -0.019095 -0.003822   
Ship Month   -0.015423 -0.016927      NaN     0.017481  0.014876 -0.011930   
Ship Year    -0.023664  0.038792      NaN     0.007030  0.012767  0.003875   

                Sales  Order Day  Order Month  Order Year  Ship Day  \
Ship Mode    0.002101   0.020068    -0.010100   -0.026663  0.031236   
Segment      0.010924   0.003919    -0.022306    0.040080  0.002013   
Country           NaN        NaN          NaN         NaN       NaN   
Postal Code -0.024002   0.017929     0.021646    0.006415  0.029721   
Region       0.000070  -0.011606     0.017303    0.012116 -0.019095   
Category     0.041375  -0.014295    -0.012946    0.004313 -0.003822   
Sales        1.000000   0.003301    -0.000134   -0.010624  0.013511   
Order Day    0.003301   1.000000    -0.031614   -0.019780  0.393263   
Order Month -0.000134  -0.031614     1.000000   -0.018278 -0.010931   
Order Year  -0.010624  -0.019780    -0.018278    1.000000 -0.012731   
Ship Day     0.013511   0.393263    -0.010931   -0.012731  1.000000   
Ship Month   0.003942  -0.062213     0.911756   -0.005323  0.019973   
Ship Year   -0.011844   0.000801    -0.001038    0.994092 -0.033120   

             Ship Month  Ship Year  
Ship Mode     -0.015423  -0.023664  
Segment       -0.016927   0.038792  
Country             NaN        NaN  
Postal Code    0.017481   0.007030  
Region         0.014876   0.012767  
Category      -0.011930   0.003875  
Sales          0.003942  -0.011844  
Order Day     -0.062213   0.000801  
Order Month    0.911756  -0.001038  
Order Year    -0.005323   0.994092  
Ship Day       0.019973  -0.033120  
Ship Month     1.000000  -0.032606  
Ship Year     -0.032606   1.000000